src/Handlers/Reply/ReplyHandler.ts

```diff
@@ -0,0 +1,71 @@
+import crypto from "crypto";
+import HttpMethods from "../../Constants/HttpMethods";
+import {ICredentials} from "../../Utils/Types";
+import {Hmac} from "../../Request/Hmac";
+import buckarooClient from "../../BuckarooClient";
+
+export class ReplyHandler {
+    private readonly data: object
+    private readonly uri?: string
+    private readonly auth_header?: string
+    private credentials: ICredentials;
+    private _isValid: boolean = false
+
+    constructor(credentials: ICredentials, data: string | object,auth_header?: string, uri?: string) {
+        if(typeof data === 'string'){
+            try {
+                this.data = JSON.parse(data)
+            } catch (e){
+                let objData = {}
+                new URLSearchParams(data).forEach((value, name)=>{
+                    objData[name] = value
+                })
+                this.data = objData
+            }
+        }else {
+            this.data = data
+        }
```

Comment:
What other data type can you expect other than a string?

Reply:
Was thinking of cases where the data is already parsed so there is no need to parse it again